src/form/__tests__/analytics.test.tsx

```diff
@@ -350,4 +339,25 @@ describe('Form Analytics', () => {
       })
     );
   });
+
+  test('forms embedded inside a Modal use Modal header for the Funnel Name', () => {
+    render(
+      <Modal header="My Modal Funnel" visible={true}>
+        <Form>
+          <Container header={<Header>Substep one</Header>}></Container>
+          <Container header={<Header>Substep two</Header>}></Container>
+        </Form>
+      </Modal>
+    );
+
+    act(() => void jest.runAllTimers());
```

Comment:
Not blocking for this PR, but I have received customer complains about analytics caused "unwrapped act" warnings. We need to more carefully consider every case where we have to wrap some updates in `act` in order to work